File: spring-data-requery/src/main/java/org/springframework/data/requery/listeners/LogbackListener.java
package org.springframework.data.requery.listeners;

import io.requery.sql.BoundParameters;
import io.requery.sql.EntityStateListener;
import io.requery.sql.StatementListener;
import lombok.extern.slf4j.Slf4j;

import java.sql.Statement;

/**
 * org.springframework.data.requery.listeners.LogbackListener
 *
 * @author debop
 */
@Slf4j
public class LogbackListener<E> implements EntityStateListener<E>, StatementListener {

    private final boolean printStatement;

    public LogbackListener() {
        this(false);
    }

    public LogbackListener(boolean printStatement) {
        this.printStatement = printStatement;
    }

    @Override
    public void postDelete(E entity) {
        log.debug("postDelete {}", entity);
    }

    @Override
    public void postInsert(E entity) {
        log.debug("postInsert {}", entity);
    }

    @Override
    public void postLoad(E entity) {
        log.debug("postLoad {}", entity);
    }

    @Override
    public void postUpdate(E entity) {
        log.debug("postUpdate {}", entity);
    }

    @Override
    public void preDelete(E entity) {
        log.debug("preDelete {}", entity);
    }

    @Override
    public void preInsert(E entity) {
        log.debug("preInsert {}", entity);
    }

    @Override
    public void preUpdate(E entity) {
        log.debug("preUpdate {}", entity);
    }

    @Override
    public void beforeExecuteUpdate(Statement statement, String sql, BoundParameters parameters) {

    }

    @Override
    public void afterExecuteUpdate(Statement statement, int count) {
        log.debug("afterExecuteUpdate count={}", count);
    }

    @Override
    public void beforeExecuteBatchUpdate(Statement statement, String sql) {
        if (statement != null) {
            log.debug("beforeExecuteBatchUpdate SQL:\n{}", sql);

            if (printStatement) {
                log.debug("{}", statement);
            }
        }
    }

    @Override
    public void afterExecuteBatchUpdate(Statement statement, int[] count) {
        log.debug("afterExecuteBatchUpdate count={}", count);
    }

    @Override
    public void beforeExecuteQuery(Statement statement, String sql, BoundParameters parameters) {
        if (statement != null) {
            if (parameters != null && !parameters.isEmpty()) {
                log.debug("beforeExecuteUpdate SQL:\n{} ({})", sql, parameters);
            } else {
                log.debug("beforeExecuteUpdate SQL:\n{}", sql);
            }

            if (printStatement) {
                log.debug("{}", statement);
            }
        }
    }

    @Override
    public void afterExecuteQuery(Statement statement) {
        log.debug("afterExecuteQuery");
    }
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/listeners/LogbackListener.kt
package org.springframework.data.requery.kotlin.listeners

import io.requery.sql.BoundParameters
import io.requery.sql.EntityStateListener
import io.requery.sql.StatementListener
import mu.KLogging
import java.sql.Statement

/**
 * LogbackListener
 *
 * @author debop@coupang.com
 * @since 18. 7. 2
 */
class LogbackListener<T> @JvmOverloads constructor(val printStatement: Boolean = false)
    : EntityStateListener<T>, StatementListener {

    companion object: KLogging()

    override fun preInsert(entity: T) {
        logger.debug { "preInsert $entity" }
    }

    override fun preUpdate(entity: T) {
        logger.debug { "preUpdate $entity" }
    }

    override fun preDelete(entity: T) {
        logger.debug { "preDelete $entity" }
    }

    override fun postDelete(entity: T) {
        logger.debug { "postDelete $entity" }
    }

    override fun postInsert(entity: T) {
        logger.debug { "postInsert $entity" }
    }

    override fun postUpdate(entity: T) {
        logger.debug { "postUpdate $entity" }
    }

    override fun postLoad(entity: T) {
        logger.debug { "postLoad $entity" }
    }

    override fun afterExecuteUpdate(statement: Statement?, count: Int) {
        logger.debug { "afterExecuteUpdate count=[$count]" }
    }

    override fun beforeExecuteBatchUpdate(statement: Statement?, sql: String?) {
        statement?.let {
            logger.debug { "beforeExecuteBatchUpdate SQL:\n$sql" }
            if(printStatement)
                logger.debug { "$statement" }
        }
    }

    override fun afterExecuteQuery(statement: Statement?) {
        logger.debug { "afterExecuteQuery" }
    }

    override fun afterExecuteBatchUpdate(statement: Statement?, count: IntArray?) {
        logger.debug { "afterExecuteBatchUpdate count=[$count]" }
    }

    override fun beforeExecuteUpdate(statement: Statement?, sql: String?, parameters: BoundParameters?) {
        if(parameters != null && !parameters.isEmpty) {
            logger.debug { "beforeExecuteUpdate SQL:\n$sql ($parameters)" }
        } else {
            logger.debug { "beforeExecuteUpdate SQL:\n$sql" }
        }

        if(printStatement)
            logger.debug { "$statement" }
    }

    override fun beforeExecuteQuery(statement: Statement?, sql: String?, parameters: BoundParameters?) {
        if(parameters != null && !parameters.isEmpty) {
            logger.debug { "beforeExecuteQuery SQL:\n$sql ($parameters)" }
        } else {
            logger.debug { "beforeExecuteQuery SQL:\n$sql" }
        }

        if(printStatement)
            logger.debug { "$statement" }
    }
}